lib/collection/src/shards/replica_set/update.rs

```diff
@@ -678,11 +711,61 @@ impl ShardReplicaSet {
             .forward_update(
                 OperationWithClockTag::from(operation),
                 wait,
+                timeout,
                 ordering,
                 hw_measurement_acc,
             ) // `clock_tag` *has to* be `None`!
             .await
     }
+
+    /// Merges successful update results from a replica set.
+    ///
+    /// If there are WaitTimeout results, the merged result will be WaitTimeout.
+    /// If there are Completed results, the merged result will be Completed.
+    /// Otherwise, the merged result will be Acknowledged.
+    fn merge_successful_update_results<F>(
+        this_peer_id: PeerId,
+        successes: &[(PeerId, UpdateResult)],
+        is_source_of_truth: F,
+    ) -> UpdateResult
+    where
+        F: Fn(PeerId) -> bool,
+    {
+        debug_assert!(!successes.is_empty());
+        debug_assert!(
+            !successes
+                .iter()
+                .any(|(_, r)| r.status == UpdateStatus::ClockRejected),
+            "ClockRejected must be handled before merging successful results"
+        );
+
+        // Aggregate status: WaitTimeout > .. > ClockRejected
+        let status = successes
+            .iter()
+            .map(|(_, res)| res.status)
+            .max_by_key(|s| s.priority())
+            .unwrap_or(UpdateStatus::Acknowledged);
+
+        // Pick a representative result to carry non-aggregated fields (operation_id, etc.).
+        // Prefer the most authoritative peer: local source-of-truth > any source-of-truth > local > any.
+        let mut result = successes
+            .iter()
+            .max_by_key(|(peer_id, _)| (is_source_of_truth(*peer_id), *peer_id == this_peer_id))
+            .map(|(_, res)| *res)
+            .expect("successes is not empty");
+
+        // Overwrite with the aggregated status across all successful replies
+        result.status = status;
+
+        // Propagate the freshest clock we observed among successful replies.
```

Comment:
I think previous implementation didn't care about it, right? I propose to keep old behaviour just in case